src/PubSub/PubSubClient.php

```diff
@@ -0,0 +1,308 @@
+     *
+     * $pubsub = new PubSubClient([
+     *     'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * // No API request yet!
+     * $topic = $pubsub->topic('my-new-topic');
+     *
+     * // This will execute an API call.
+     * echo $topic->info()['name']; // `projects/my-awesome-project/topics/my-new-topic`
+     * ```
+     *
+     * @param  string $name The topic name
+     * @param  array  $info Information about the topic. Used internally to
+     *     populate topic objects with an API result. Should be
+     *     an instance of[https://cloud.google.com/pubsub/reference/rest/v1/projects.topics#Topic](Topic).
+     * @return Topic
+     */
+    public function topic($name, array $info = null)
+    {
+        return new Topic($this->connection, $name, $this->projectId, $info);
+    }
+
+    /**
+     * Get a list of the topics registered to your project.
+     *
+     * ```php
+     * use Google\Cloud\PubSub\StorageClient;
+     *
+     * $pubsub = new PubSubClient([
+     *     'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * $topics = $pubsub->topics();
+     * foreach ($topics as $topic) {
+     *      $info = $topic->info();
+     *      echo $info['name']; // `projects/my-awesome-project/topics/<topic-name>`
+     * }
+     * ```
+     *
+     * @param  array     $options Configuration options
+     * @return Generator An array of Topics
+     */
+    public function topics(array $options = [])
+    {
+        $options['pageToken'] = null;
+
+        do {
+            $response = $this->connection->listTopics($options + [
+                'project' => $this->formatName('project', [$this->projectId])
+            ]);
+
+            foreach ($response['topics'] as $topic) {
+                yield $this->topic($topic['name'], $topic);
+            }
+
+            // If there's a page token, we'll request the next page.
+            $options['pageToken'] = isset($response['nextPageToken'])
+                ? $response['nextPageToken']
+                : null;
+        } while ($options['pageToken']);
+    }
+
+    /**
+     * Create a Subscription. If the subscription does not exist, it will be
+     * created.
+     *
+     * Use {@see PubSubClient::subscription()} to create a subscription object
+     * without any API requests. If the topic already exists, an exception will
+     * be thrown. When in doubt, use {@see PubSubClient::subscription()}.
+     *
+     * ```php
+     * use Google\Cloud\PubSub\StorageClient;
+     *
+     * $pubsub = new PubSubClient([
+     *     'projectId' => 'my-awesome-project'
+     * ]);
+     *
+     * // Create a subscription. If it doesn't exist in the API, it will be created.
+     * $subscription = $pubsub->subscribe('my-new-subscription', 'my-topic-name');
+     * ```
+     *
+     * @param  string       $name      A subscription name
+     * @param  string       $topicName The topic to attach to
+     * @param  array        $options   Configuration options
```

Comment:
It would be nice to show an example of supplying PUSH config (i.e. `push-endpoint`) here.